CardsCashCasino/Data/Card.cs

```diff
@@ -0,0 +1,261 @@
+        {
+            BlackjackValueAttribute blackjackValue = _value.GetAttribute<BlackjackValueAttribute>()!;
+            if (blackjackValue.HasTwoValues)
+                return -1;
+            else
+                return (int)blackjackValue.SecondaryValue!;
+        }
+
+        /// <summary>
+        /// Returns the poker value.
+        /// </summary>
+        public int GetPokerValue()
+        {
+            return _value.GetAttribute<PokerValueAttribute>()!.PokerValue;
+        }
+
+        /// <summary>
+        /// The equals operator between two card objects.
+        /// </summary>
+        public static bool operator ==(Card card1, Card card2) { return card1._value == card2._value && card1._suit == card2._suit; }
+
+        /// <summary>
+        /// The not equals operator between two card objects.
+        /// </summary>
+        public static bool operator !=(Card card1, Card card2) { return card1._value != card2._value || card1._suit != card2._suit; }
+
+        /// <summary>
+        /// <inheritdoc/>
+        /// </summary>
+        public override bool Equals(object? obj) { return ((obj is not null) && (obj is Card)) ? (this == (Card)obj) : false; }
+
+        /// <summary>
+        /// <inheritdoc/>
+        /// </summary>
+        public override int GetHashCode() { return base.GetHashCode(); }
+    }
+
+    /// <summary>
+    /// The enumeration value for the card's suit.
+    /// </summary>
+    public enum Suit
+    {
+        CLUBS,
+        DIAMONDS,
+        HEARTS,
+        SPADES
+    }
+
+    /// <summary>
+    /// The enumeration value for the card's value.
+    /// Contains two attributes: Blackjack and Poker Value.
+    /// </summary>
+    public enum Value
+    {
+        [BlackjackValue(11, 1)]
+        [PokerValue(1)]
+        ACE,
+        
+        [BlackjackValue(2)]
+        [PokerValue(2)]
+        TWO,
+        
+        [BlackjackValue(3)]
+        [PokerValue(3)]
+        THREE,
+        
+        [BlackjackValue(4)]
+        [PokerValue(4)]
+        FOUR,
+        
+        [BlackjackValue(5)]
+        [PokerValue(5)]
+        FIVE,
+        
+        [BlackjackValue(6)]
+        [PokerValue(6)]
+        SIX,
+
+        [BlackjackValue(7)]
+        [PokerValue(7)]
+        SEVEN,
+
+        [BlackjackValue(8)]
+        [PokerValue(8)]
+        EIGHT,
+
+        [BlackjackValue(9)]
+        [PokerValue(9)]
+        NINE,
+
+        [BlackjackValue(10)]
+        [PokerValue(10)]
+        TEN,
+
+        [BlackjackValue(10)]
+        [PokerValue(11)]
+        JACK,
+
+        [BlackjackValue(10)]
+        [PokerValue(12)]
+        QUEEN,
+
+        [BlackjackValue(10)]
+        [PokerValue(13)]
+        KING
+    }
+
+    /// <summary>
+    /// Enumeration attribute for the Blackjack value.
+    /// Can either be a single integer, or two integers.
+    /// </summary>
+    public class BlackjackValueAttribute : Attribute
+    {
+        internal BlackjackValueAttribute(int value)
```

Comment:
Does the access level of enumeration attributes need to be internal? This isn't so much of a critique as a point of information. Everything I've looked up doesn't explain to me why this level of protection is needed is opposed to private or protected.

Reply:
It's just the level of protection I found on the StackOverflow page I copied it from. I don't think it needs to be different, but it does not matter so I am just going to keep it as is.